samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/core/BaseFragment.java

```diff
@@ -0,0 +1,50 @@
+package io.customer.android.sample.java_layout.ui.core;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.EmptySuper;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.viewbinding.ViewBinding;
+
+import io.customer.android.sample.java_layout.SampleApplication;
+import io.customer.android.sample.java_layout.di.ApplicationGraph;
+
+public abstract class BaseFragment<VB extends ViewBinding> extends Fragment {
+
+    protected VB binding;
+    protected ApplicationGraph applicationGraph;
```

Comment:
Do you believe that holding a strong reference to the dependency injection graph in a Fragment will increase the risk of a memory leak from occurring?

Reply:
Oh yeah, thanks for pointing out. Looking at [viewbinding docs](https://developer.android.com/topic/libraries/view-binding#fragments), we need to clear binding reference in `onDestroy`. I've [pushed the changes](https://github.com/customerio/customerio-android/pull/200/commits/289ea62f7051f5312c40636e4aba558ea117a66f) for it.